### ISSUE TITLE
[General] newserv QST files appear to be encoded incorrectly

### ISSUE THREAD

**Issue by Matt-Swift** [fuzziqersoftware/newserv#136]
Test files:

[quest34_e.qst.zip](https://github.com/fuzziqersoftware/newserv/files/12837423/quest34_e.qst.zip)
[newserv34.qst.zip](https://github.com/fuzziqersoftware/newserv/files/12837424/newserv34.qst.zip)

The above 2 files are my unofficial translation of the quest "Black Paper's Dangerous Deal 2" for Blue Burst. `quest34_e.qst` is a file generated by SCHTHack's Qedit, and `newserv34.qst` is a file created from decoding `quest34_e.qst`, then encoding it back into QST (with the option --bb).

`newserv34.qst` when put into system/quests will spit out the following error:
```
W 261978 2023-10-07 10:10:01 - [StaticGameData] Failed to index quest file q034-1p-bb.qst (invalid command in qst file)
```
`quest34_e.qst` works, for what it is worth.

The file is also unreadable by Qedit. I also noticed differences in the header, per below:
![image](https://github.com/fuzziqersoftware/newserv/assets/33199242/3998b04a-eebf-4600-8e5c-1d8043b2f08d)
![image](https://github.com/fuzziqersoftware/newserv/assets/33199242/7b6ac13b-dc74-437d-b601-6b8e97f8b8f4)

I also attempted the same with one of newserv's default quests (q011-1p-bb.bin/.dat), and the same issue occurs.

**Comment by fuzziqersoftware** [comment]
This should be fixed by f77e2180. For BB QST files, the format apparently expects 4 padding bytes after each write data command, which newserv was not generating. We also weren't setting the file size in the open file commands correctly.

**Comment by fuzziqersoftware** [comment]
The QST format is bad and underspecified and I hate that essentially every other tool expects it.

This should be fixed now.
